packages/snaps-sdk/src/jsx/validation.ts

```diff
@@ -339,6 +341,20 @@ export const InputStruct: Describe<InputElement> = elementWithSelectiveProps(
   },
 );
 
+/**
+ * A struct for the {@link AddressInputElement} type.
+ */
+export const AddressInputStruct: Describe<AddressInputElement> = element(
+  'AddressInput',
+  {
+    name: string(),
+    chainId: CaipChainIdStruct as unknown as Struct<CaipChainId, CaipChainId>,
```

Comment:
Why is this necessary? Seems like an issue with `CaipChainIdStruct`.

Reply:
Had to do this because of the way the struct is defined with `definePattern` I think, the error is:  

```
Type 'Struct<{ key: Key | null; type: Struct<Name, Name>; props: { name: string; chainId: `${string}:${string}`; disabled?: boolean | undefined; value?: string | undefined; placeholder?: string | undefined; }; }, { ...; }>' is not assignable to type 'Describe<SnapElement<AddressInputProps, "AddressInput">>'.
  The types of 'schema.props.schema.chainId.schema' are incompatible between these types.
    Type 'null' is not assignable to type '`${string}:${string}`'
```

Reply:
Yup we have this error everytime we use `CaipChainIdStruct` :(

Reply:
Fixed the error, needed to update `StructSchema` to have a case for `CaipChainId`.